templates/common/aws/files/usr-local-bin-aws-kubelet-nodename.yaml

```diff
@@ -0,0 +1,21 @@
+mode: 0755
+path: "/usr/local/bin/aws-kubelet-nodename"
+contents:
+  inline: |
+    #!/bin/bash
+    set -e -o pipefail
+
+    NODECONF=/etc/systemd/system/kubelet.service.d/20-aws-node-name.conf
+
+    if [ -e "${NODECONF}" ]; then
+        echo "Not replacing existing ${NODECONF}"
+        exit 0
+    fi
+
+    # For compatibility with the AWS in-tree provider
+    # Set node name to be instance name instead of the default FQDN hostname
+    name=$(curl -s http://169.254.169.254/2016-09-02/meta-data/local-hostname)
```

Comment:
would be nice to have a comment referring to the aws docs that reference this ip address, just so it's less confusing to a future person looking at this.

Reply:
Done